.internal-ci/util/set_tokens_config_version.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+# Copyright (c) 2018-2023 The MobileCoin Foundation
+
+# Select the correct tokens file to use based on release version.
+
+set -eu
+
+location=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+# shellcheck source=.shared_functions
+source "${location}/.shared_functions"
+
+network=$(get_network_tier "${1}")
+major=$(get_major_version "${1}")
+config=""
+
+echo "Found network ${network}" >&2
+echo "Found major version ${major}" >&2
+# 0 - dev use V2
+# 1|2|3 - use V1 - note 1 doesn't consume
+# 4 or greater use v2
+case "${network}" in
+    test)
+        if [[ ${major} -eq 0 ]]
+        then
+            echo "Using TEST_TOKENS_CONFIG_V2_JSON" >&2
+            config=${TEST_TOKENS_CONFIG_V2_JSON}
+        elif [[ ${major} -ge 1 ]] && [[ ${major} -le 3 ]]
+        then
+            echo "Using TEST_TOKENS_CONFIG_V1_JSON" >&2
+            config=${TEST_TOKENS_CONFIG_V1_JSON}
+        elif [[ ${major} -ge 4 ]]
+        then
+            echo "Using TEST_TOKENS_CONFIG_V2_JSON" >&2
+            config=${TEST_TOKENS_CONFIG_V2_JSON}
+        else
+            echo "Major version is invalid? ${1} ${major}" >&2
+            exit 1
+        fi
+    ;;
+    main)
+        if [[ ${major} -eq 0 ]]
+        then
+            echo "Using MAIN_TOKENS_CONFIG_V2_JSON" >&2
+            config=${MAIN_TOKENS_CONFIG_V2_JSON}
+        elif [[ ${major} -ge 1 ]] && [[ ${major} -le 3 ]]
+        then
+            echo "Using MAIN_TOKENS_CONFIG_V1_JSON" >&2
+            config=${MAIN_TOKENS_CONFIG_V1_JSON}
+        elif [[ ${major} -ge 4 ]]
+        then
+            echo "Using MAIN_TOKENS_CONFIG_V2_JSON" >&2
+            config=${MAIN_TOKENS_CONFIG_V2_JSON}
+        else
+            echo "Major version is invalid? ${1} ${major}" >&2
+            exit 1
+        fi
+    ;;
+    dev)
+       if [[ ${major} -eq 0 ]]
+        then
+            echo "Using DEV_TOKENS_CONFIG_V2_JSON" >&2
+            config=${DEV_TOKENS_CONFIG_V2_JSON}
+        elif [[ ${major} -ge 1 ]] && [[ ${major} -le 3 ]]
+        then
+            echo "Using DEV_TOKENS_CONFIG_V1_JSON" >&2
+            config=${DEV_TOKENS_CONFIG_V1_JSON}
+        elif [[ ${major} -ge 4 ]]
+        then
+            echo "Using DEV_TOKENS_CONFIG_V2_JSON" >&2
+            config=${DEV_TOKENS_CONFIG_V2_JSON}
+        else
+            echo "Major version is invalid? ${1} ${major}" >&2
+            exit 1
+        fi
+    ;;
+    *)
+        echo "ERROR: Unknown network ${network}" >&2
+        exit 1;
+    ;;
+esac
+
+echo "${config}"
```

Comment:
:upside_down_face: the suggestion might be too indirect of a solution
`get_network_tier` returns `dev` for the default so the unknown network case won't be hit
```suggestion
if [[ ${major} -eq 0 ]]
then
    version="V2"
elif [[ ${major} -ge 1 ]] && [[ ${major} -le 3 ]]
then
    version="V1"
elif [[ ${major} -ge 4 ]]
then
    version="V2"
else
    echo "Major version is invalid? ${1} ${major}" >&2
    exit 1
fi

token_json="${network^^}_TOKENS_CONFIG_${version}_JSON"
echo "Using ${token_json}" >&2
echo "${!token_json}"
```

Reply:
I was debating if it would be more long term supportable (understandable by someone else in 6 months)  to use the longer case/if or construct the variable name.  I think I'll take your suggestions here and tweak them a bit and add some comments for the parameter expansion bits.  I can never remember what they do with out googling.